.travis.yml

```diff
@@ -3,26 +3,37 @@ python:
-  - echo "deb http://packages.spectrum.im/spectrum2/ xenial main" | sudo tee -a /etc/apt/sources.list
-  - echo "deb http://swift.im/packages/ubuntu/xenial beta main" | sudo tee -a /etc/apt/sources.list
-  - echo "deb http://packages.prosody.im/debian $(lsb_release -sc) main" | sudo tee -a /etc/apt/sources.list
-  - sudo apt-get update -qq
-  - sudo apt-get purge chromium-browser
-  - sudo apt-get install -y --force-yes build-essential prosody ngircd python-sleekxmpp libswiften-dev libprotobuf-dev protobuf-compiler libpurple-dev libglib2.0-dev libdbus-glib-1-dev liblog4cxx10-dev libpopt-dev libboost-dev libboost-signals-dev libboost-system-dev libboost-thread-dev libboost-locale-dev libboost-filesystem-dev libboost-program-options-dev libboost-regex-dev libboost-date-time-dev libcppunit-dev libcommuni-dev libminiupnpc-dev libnatpmp-dev
-  - sudo killall lua5.1
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+      export DEBIAN_FRONTEND=noninteractive;
+      export APT_LISTCHANGES_FRONTEND=none;
+      echo force-confold | sudo tee -a /etc/dpkg/dpkg.cfg;
+      curl -k https://swift.im/keys/packages.key | sudo apt-key add -;
+      echo 'deb http://us.archive.ubuntu.com/ubuntu xenial           main restricted universe multiverse' | sudo tee    /etc/apt/sources.list.d/xenial.list > /dev/null;
+      echo 'deb http://us.archive.ubuntu.com/ubuntu xenial-backports main restricted universe multiverse' | sudo tee -a /etc/apt/sources.list.d/xenial.list > /dev/null;
+      echo 'deb http://us.archive.ubuntu.com/ubuntu xenial-security  main restricted universe multiverse' | sudo tee -a /etc/apt/sources.list.d/xenial.list > /dev/null;
+      echo 'deb http://us.archive.ubuntu.com/ubuntu xenial-updates   main restricted universe multiverse' | sudo tee -a /etc/apt/sources.list.d/xenial.list > /dev/null;
+      echo "deb http://packages.spectrum.im/spectrum2/ xenial main" | sudo tee -a /etc/apt/sources.list;
+      echo "deb http://swift.im/packages/ubuntu/xenial beta main" | sudo tee -a /etc/apt/sources.list;
+      echo "deb http://packages.prosody.im/debian $(lsb_release -sc) main" | sudo tee -a /etc/apt/sources.list;
+      sudo apt-get update -qq;
+      sudo apt-get purge chromium-browser;
+      sudo apt-get install -y --force-yes build-essential prosody ngircd python-sleekxmpp libswiften-dev libprotobuf-dev protobuf-compiler libpurple-dev libglib2.0-dev libdbus-glib-1-dev liblog4cxx10-dev libpopt-dev libboost-dev libboost-signals-dev libboost-system-dev libboost-thread-dev libboost-locale-dev libboost-filesystem-dev libboost-program-options-dev libboost-regex-dev libboost-date-time-dev libcppunit-dev libcommuni-dev libminiupnpc-dev libnatpmp-dev;
+      sudo killall lua5.1;
+    else
+      brew update;
+      brew install prosody ngircd libswiften protobuf pidgin log4cxx popt boost cppunit;
+    fi
 install: "pip install --user sleekxmpp"
 script:
-  - cmake -DCMAKE_BUILD_TYPE=Debug -DENABLE_TESTS=ON . && make && make extended_test
```

Comment:
`test` just run library tests, while `extended_test` spin up prosody/ngircd and run some integration tests